students/Ben_freeman/lesson04/mailroom2.py

```diff
@@ -0,0 +1,88 @@
+#Donor dict
+donor_list={
+"Ahmed":[123,456,789],
+"Sally": [10,11],
+"Bob": [12],
+"Vladamir": [13.14,1516.17],
+"Santa": [181920.21,22,23.24],
+}
+
+
+#send thanks funct
+def send_thanks():
+    while True:
+        query=input("Please enter a Full Name, enter 'list' to see a list of current donors, or quit to exit: ")
+        if query in donor_list:
+            new_donate=float(input("How much would you like to donate: \n"))
+            donor_list[query].append(new_donate)
+            print(f"\nThank you {query} for your donation of ${new_donate}, your donations make the work of the 'American society for taking donations' possible.\n\n"
+                "Sincearly,\n\n"
+                "A low paid intern\n")
+            break
+        elif query=="list":
+            print('\n'.join(donor_list))
+        elif query.lower()=="quit":
+            return
+        elif query not in donor_list:
+            donor_list[query]=[]
+            new_donate=float(input("\nHow much would you like to donate: \n"))
+            donor_list[query].append(new_donate)
+            print(f"\nThank you {query} for your donation of ${new_donate}, your donations make the work of the 'American society for taking donations' possible.\n\n"
+                "Sincearly,\n\n"
+                "A low paid intern\n")
+            break
+        
+
+#making a sortable list
+def sorting_function(dlist):
+    mrgl={}
+    for i in dlist:
+        mrgl[i]=[sum(dlist[i]),len(dlist[i]),sum(dlist[i])/len(dlist[i])]
+    flist=[]
+    for keys, values in mrgl.items():
+        values.append(keys)
+        flist.append(values)
+    return sorted(flist, key=lambda flist:flist[0], reverse=True)
```

Comment:
I'm having a hard time understanding above function. 

Typically, you would just create a list of tuples, each tuple containing the summary info for the report.

Reply:
This is my hamfisted way of turning my dictionary into a list which is then sorted. 
```
    mrgl={}
    for i in dlist:
        mrgl[i]=[sum(dlist[i]),len(dlist[i]),sum(dlist[i])/len(dlist[i])]
```
collects all of the information that we want for each donor and assigns it to that donor

```    flist=[]
    for keys, values in mrgl.items():
        values.append(keys)
        flist.append(values)
    return sorted(flist, key=lambda flist:flist[0], reverse=True)
```
this takes the temporary dictionary that we just produced and merges the numerical values and donor name together to form a list which we then sort. 

Looking at it again it seem like there is a lot of unnecessary code here and it could probably be shortened to something like 

```
def sorting_function(dictionary):
    temp_list=[]
    for keys, values in dictionary.items():
        temp_list.append([sum(values),keys,len(values),sum(values)/len(values)])
    return sorted(temp_list, reverse=True)
```

Is there a benefit to using a list of tuples over a list of lists or would this be sufficient?

Reply:
your condensed version is exactly what I was suggesting.

The main reason you would use a tuple over a list is if you want to be clear and sure that a list item will not be modified by the consumer of the data. You can use either but a list of tuples is technically more correct.